packages/viewer/src/components/viewer/viewer.tsx

```diff
@@ -163,6 +168,11 @@ export class Viewer {
    */
   @Event() public tokenExpired!: EventEmitter<void>;
 
+  /**
+   * Emits an event when the connection status changes for the viewer
+   */
+  @Event() public connectionChange!: EventEmitter<ConnectionStatus>;
```

Comment:
curious to get @danschultz 's thoughts on this as well, I think I like the interface of having specific events for the different connection statuses vs a single event where I need to check the status in the handler to handle the connection status I'm looking for. but probably comes down to whether we expect developers to more frequently want to run code whenever the connection status changes or more often when the connection status is a certain value.

Reply:
My personal opinion if I were a consumer would be to subscribe to one location.

Reply:
If we're to do this, then we should turn `ConnectionStatus` into an intersection type. Some fields don't make any sense in a `connecting` or `disconnected` state, and having these additional types provides type safety within the context of our TS developers. Example:

```ts
interface ConnectedStatus {
  jwt: string;
  status: 'connected';
}

interface ConnectingStatus {
  status: 'connecting';
}

interface DisconnectedStatus {
  status: 'disconnected';
}

type ConnectionStatus = ConnectingStatus | ConnectedStatus | DisconnectedStatus;
```

Reply:
> but probably comes down to whether we expect developers to more frequently want to run code whenever the connection status changes or more often when the connection status is a certain value.

I think there'll be consumers who'll want to know the `connected` and `disconnected` state in a pair. Use-case would be to show some status to the user if the viewer is connected or not. FWIW, the browser's connection status APIs use a single event. https://developer.mozilla.org/en-US/docs/Web/API/Network_Information_API